server2/routes/signupRouter.js

```diff
@@ -15,30 +17,39 @@ router.route('/')
     })
     .post((req, res, next) => {
         const newUser = req.body;
-        //User.sync()
+        // User.sync()
         Promise.all([isDuplicateEmail(newUser), isDuplicateUser(newUser)])
             .then((newUser) => {
                 const saltRounds = 10;
-                const salt = bcrypt.genSaltSync(saltRounds)
-                const hash = bcrypt.hashSync(req.body.password, salt);
-                
-                const test_user = User.create({
+                const salt = bcrypt.genSaltSync(saltRounds);
+                const hash = bcrypt.hashSync(req.body.password, saltRounds);
+
+                const new_user = {
                     id: idGen(),
-                    username: req.body.username,
+                    name: req.body.username,
                     email: req.body.email,
                     salt: salt,
                     hash: hash,
                     access_token: generateAccessToken(req.body.username),
```

Comment:
(after looking back from verifyEmail function) salt, hash, and access_token will be totally useless since it will neither be used in the email, not save to the database. Then why you generate these 3 fields?